storage/aptosdb/src/fast_sync_aptos_db.rs

```diff
@@ -0,0 +1,563 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    backup::backup_handler::DbState,
+    metrics::{
+        BACKUP_EPOCH_ENDING_EPOCH, BACKUP_STATE_SNAPSHOT_LEAF_IDX, BACKUP_STATE_SNAPSHOT_VERSION,
+        BACKUP_TXN_VERSION,
+    },
+    AptosDB,
+};
+use anyhow::{anyhow, ensure, Context, Result};
+use aptos_config::config::{BootstrappingMode, NodeConfig};
+use aptos_crypto::HashValue;
+use aptos_storage_interface::{
+    cached_state_view::ShardedStateCache, state_delta::StateDelta, DbReader, DbWriter,
+    ExecutedTrees, FastSyncStatus, Order, StateSnapshotReceiver,
+};
+use aptos_types::{
+    account_config::NewBlockEvent,
+    contract_event::{ContractEvent, EventWithVersion},
+    epoch_change::EpochChangeProof,
+    epoch_state::EpochState,
+    event::EventKey,
+    ledger_info::LedgerInfoWithSignatures,
+    proof::{
+        AccumulatorConsistencyProof, SparseMerkleProof, SparseMerkleProofExt,
+        SparseMerkleRangeProof, TransactionAccumulatorRangeProof, TransactionAccumulatorSummary,
+        TransactionInfoWithProof,
+    },
+    state_proof::StateProof,
+    state_store::{
+        state_key::StateKey,
+        state_key_prefix::StateKeyPrefix,
+        state_storage_usage::StateStorageUsage,
+        state_value::{StateValue, StateValueChunkWithProof},
+        table::{TableHandle, TableInfo},
+        ShardedStateUpdates,
+    },
+    transaction::{
+        AccountTransactionsWithProof, Transaction, TransactionInfo, TransactionListWithProof,
+        TransactionOutputListWithProof, TransactionToCommit, TransactionWithProof, Version,
+    },
+    write_set::WriteSet,
+};
+use move_core_types::account_address::AccountAddress;
+use std::sync::{Arc, RwLock};
+
+const SECONDARY_DB_DIR: &str = "fast_sync_secondary";
+
+/// This is a wrapper around [AptosDB] that is used to bootstrap the node for fast sync mode
+pub struct FastSyncStorageWrapper {
+    // main is used for normal read/write or genesis data during fast sync
```

Comment:
nit: can we update this comment? Also, do you think it makes sense to rename these now? e.g., `temporary_db_with_genesis` and `real_db_for_fast_sync`?

Reply:
updated